src/v/cluster/partition_balancer_planner.cc

```diff
@@ -453,6 +466,16 @@ partition_balancer_planner::plan_reassignments(
 
     if (_topic_table.has_updates_in_progress()) {
         get_unavailable_node_movement_cancellations(result.cancellations, rrs);
+        if (!result.cancellations.empty()) {
+            result.status
+              = partition_balancer_planner::status::cancellations_planned;
+        }
+        return result;
+    }
+
+    if (!all_reports_received(rrs)) {
```

Comment:
should we skip calculating nodes with disk constraint violations when we haven't yet received all reports? I guess the correct order would be:
* calculate unavailable nodes
* (maybe) calculate cancellations
* if not all reports received, bail out
* else initialize from the health report and continue

Reply:
In that order we won't see full nodes in status